hawc/apps/lit/serializers.py

```diff
@@ -196,3 +196,21 @@ def bulk_create_tags(self):
             models.ReferenceTags.objects.bulk_create(new_tags)
 
             models.Reference.delete_cache(assessment_id)
+
+
+class ReferenceSerializer(serializers.ModelSerializer):
+    tags = serializers.ListField(write_only=True)
+
+    def update(self, instance, validated_data):
+
+        if "tags" in validated_data:
```

Comment:
tags are popped here and then never used; does this mean they're not saved?

Reply:
the `instance.tags.set` updates and saves the reference with the given tags, so at that point they're popped since the operation is done / that key is unnecessary in the next steps. I'll add some comments clarifying it

Reply:
should we decorate w/ `@transaction.atomc()` or something similar since it sounds like there are multiple db operations here?